pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -422,6 +424,23 @@ func CreateGCECloud(config *CloudConfig) (*GCECloud, error) {
 		subnetURL = config.SubnetworkURL
 	} else if config.SubnetworkName != "" {
 		subnetURL = gceSubnetworkURL(config.ApiEndpoint, netProjID, config.Region, config.SubnetworkName)
+	} else {
+		n, err := getNetwork(service, netProjID, lastComponent(networkURL))
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve network %v, err: %v", networkURL, err)
+		}
+
+		// Legacy networks have a non-empty IPv4Range
+		if len(n.IPv4Range) > 0 {
+			// If legacy network, accept missing subnet
+			isLegacyNetwork = true
+		} else {
+			// Try to find the subnet in the list of subnets
+			subnetURL = findSubnetForRegion(n.Subnetworks, config.Region)
```

Comment:
@bowei Originally, I threw an error if no subnet is specified and n.AutoCreateSubets is false. Given what we just saw for auto networks converted to manual networks, I removed the error.

Reply:
Let me know if you disagree or have a better idea.